### DISCUSSION TITLE
How to format date column in DetailsList?

### DISCUSSION THREAD

**Discussion by kthirapatpibul** [microsoft/powercat-creator-kit#120] [Q&A] [answered]
By default, the date column in a Fluent DetailsList show both date and time, how do I format the date so that it only show the date? Also, it seems that null date are being displayed as 12/31/1969, how do I make it show "blank"?

**Comment by scottdurow** [comment, marked as answer]
Hi @kthirapatpibul,
You can use `AddColumns`. E.g.
1. Edit the `Items` property to include an additional column that formats your date:
```
AddColumns(
Accounts,
"ColDate",
Text('createdon',"yyyy-mm")
)
```
2. Add the new date column to the Fields property by clicking on `Items`
3. Update your columns definition to display the 'ColDate' property - but set the `ColSortBy` property to the original date so that you can sort by the real date field.
4. If you were using automatic sorting - you will now need to change to use manual sorting as described by https://learn.microsoft.com/en-us/power-platform/guidance/creator-kit/detailslist#manual-sorting

Hope this helps!

**Reply by craigyh** [reply]
Hi Scott,
This is exactly what I need. I already have a filter statement in the Items property and I can't get the syntax right to include both the filter and the add column statement. If you could let me know how to do this that would be great.

Kind Regards, Craig

**Comment by knyklu** [comment]
Hi @scottdurow,
in Yours solution we will get problems with delegation limits. Is any other way?

**Reply by e-gallis** [reply]
I confirm that there is going to be a delegation issue if the DetailsList component is directly connected to the datasource (and not a collection) as if we scroll down past the 2000 max records, the formatted 'ColDate' column will then be empty...
We would really need a property to set the display format for a column.
Is there any chance we could have this new property?

**Reply by scottdurow** [reply]
I agree - the DetailsList does need a format string property for date columns (and numeric) - in the mean time you can use the delegation approach I show here https://youtu.be/zfyoBLbPvoQ (around the 19:00) mark

**Reply by knyklu** [reply]
> I agree - the DetailsList does need a format string property for date columns (and numeric) - in the mean time you can use the delegation approach I show here https://youtu.be/zfyoBLbPvoQ (around the 19:00) mark

Hi, I did according to Yours tutorial, but after adding the AddColumns to a source of galPaging:
```
AddColumns(
    Switch(
        ctxSortCol;`
.
.
.

 "castom_column";
    Text(
        cr574_data_koniec;
        "dd-mm-yy hh:mm"
    )
```
the delegation nightmare has back.

**Reply by yaseralhosani** [reply]
I use date only and working totally fine so far for around 2k records .

**Reply by knyklu** [reply]
> I use date only and working totally fine so far for around 2k records .

With Addcolumns? Check the "Data row limit..." and compare with yours number of records.

**Reply by yaseralhosani** [reply]
> > I use date only and working totally fine so far for around 2k records .
> 
> 
> 
> With Addcolumns? Check the "Data row limit..." and compare with yours number of records.

I'm currently out of the office. Yes with AddColumns, I keep at 2000 row limit. And records is around 650-800. It was more than 2000k records and I redesign tables to reduce duplicate records.

Sometimes when I'm working in coding, no delegation when I use AddColumns(), With() as well as collection, just you need to try different techniques. I would suggest you to try out with collection. If still facing same issue probably data type issue.

if still facing same issue then try add new column for date only, do not delete that column if you are already using that column in production.

**Reply by knyklu** [reply]
I explain why I need extra column. I'm try to prepare some kind of time tracker app for Production Dept.
The dataverse table has columns
Data_start (date and time)
Data_finish(data and time)
Operation_time (this field is a calculted: DiffInMinutes).
 With data source as a raw table from Dataverse I got in detail list some value in Data_finish where fiels is empty. 
Thats the only reason*.
Maybe there is some othere way to avoid this issue. I dont know.
![image](https://github.com/microsoft/powercat-creator-kit/assets/134507744/74b14300-a076-4f3e-ba65-0afecec23063)

*Of course, it would be nice if I could fully customize displaied data.

**Reply by yaseralhosani** [reply]
There is a solution, you need to try different techniques. Usually collection solve some delegations try below.

ClearCollect(
    colTracker, 
          ForAll(
                  Tacker,
                  {
                     DateStart: Data_start,
                     Datefinish: Data_finish,
                     Operationtime: …. 
                  }
          )
)


Then add colTracker into Detailslist.item

**Comment by JuanFMartinezH** [comment]
Hey, keep in mind that the approach of @scottdurow is subjected to the delegation limit.

Another approach I used is to have another column in the dataverse table that keeps track of the formatted date. Unfortunately, I was unable to use a calculated column for this due to limitation on the functions in this type of columns. So, the value should be updated manually, anytime the date changes, either by using a Power Automate flow (Premium) or by updating the value any time you change the original date column in your app.

In the detailslist you would then reference this formatted column instead of your original date column and the query can be then delegated to dataverse.

**Reply by scottdurow** [reply]
You might be able to use the technique I show here - https://youtu.be/zfyoBLbPvoQ
